runner_manager/backend/aws.py

```diff
@@ -27,37 +27,32 @@ def create(self, runner: Runner) -> Runner:
         labels: Dict[str, str] = {}
         if self.manager:
             labels["runner-manager"] = self.manager
+        self.instance_config.subnet_id = self.config.subnet_id
         instance_resource: Dict = self.instance_config.configure_instance(runner)
         try:
             instance = self.client.run_instances(**instance_resource)
-        except ClientError as e:
-            raise e
-        try:
             runner.instance_id = instance["Instances"][0]["InstanceId"]
-        except Exception as e:
-            raise e
-        # Wait for instance to be running
-        try:
+            # Wait for instance to be running
             waiter = self.client.get_waiter("instance_running")
             waiter.wait(InstanceIds=[runner.instance_id])
         except Exception as e:
+            log.error(e)
             raise e
         return super().create(runner)
 
     def delete(self, runner: Runner):
         """Delete a runner."""
-        try:
-            if runner.instance_id:
+        if runner.instance_id:
+            try:
                 self.client.terminate_instances(InstanceIds=[runner.instance_id])
-        except ClientError as e:
-            error = e.response.get("Error", {})
-            if error.get("Code") == "InvalidInstanceID.NotFound":
-                pass
-            if error.get("Code") == "InvalidInstanceID.Malformed":
-                log.error(f"Instance {runner.instance_id} malformed.")
-                pass
-            else:
-                raise e
+            except ClientError as e:
+                error = e.response.get("Error", {})
+                if error.get("Code") == "InvalidInstanceID.NotFound":
+                    log.error(f"Instance {runner.instance_id} not found.")
+                elif error.get("Code") == "InvalidInstanceID.Malformed":
+                    log.error(f"Instance {runner.instance_id} malformed.")
```

Comment:
Ensure the delete function from the parent gets called otherwise we face the same issue as with the gcloud backend where it would be stuck in a infinite loop.
```suggestion
                if error.get("Code") == "InvalidInstanceID.NotFound":
                    log.error(f"Instance {runner.instance_id} not found.")
                    return super().delete(runner)
                elif error.get("Code") == "InvalidInstanceID.Malformed":
                    log.error(f"Instance {runner.instance_id} malformed.")
                    return super().delete(runner)
```

Reply:
Problem was not addressed. We need to ensure that the runner was deleted on the database side on some situations.